From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 19 Dec 2015 19:10:03 +0000
Subject: mm: Fix missing #include in <linux/mmdebug.h>
Forwarded: http://mid.gmane.org/20151219203034.GR28542@decadent.org.uk

The various VM_WARN_ON/VM_BUG_ON macros depend on those defined by
<linux/bug.h>.  Most users already include those, but not all; for
example:

  CC      arch/arm64/xen/../../arm/xen/grant-table.o
In file included from arch/arm64/include/../../arm/include/asm/xen/page.h:5:0,
                 from arch/arm64/include/asm/xen/page.h:1,
                 from include/xen/page.h:28,
                 from arch/arm64/xen/../../arm/xen/grant-table.c:33:
arch/arm64/include/asm/pgtable.h: In function 'set_pte_at':
arch/arm64/include/asm/pgtable.h:281:3: error: implicit declaration of function 'BUILD_BUG_ON_INVALID' [-Werror=implicit-function-declaration]
   VM_WARN_ONCE(!pte_young(pte),

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 include/linux/mmdebug.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/mmdebug.h b/include/linux/mmdebug.h
index 877ef22..772362a 100644
--- a/include/linux/mmdebug.h
+++ b/include/linux/mmdebug.h
@@ -1,6 +1,7 @@
 #ifndef LINUX_MM_DEBUG_H
 #define LINUX_MM_DEBUG_H 1
 
+#include <linux/bug.h>
 #include <linux/stringify.h>
 
 struct page;
